File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/ui/BaseActivity.java
package at.tugraz.tc.cyfile.ui;

import android.support.v7.app.AppCompatActivity;

import at.tugraz.tc.cyfile.CyFileApplication;
import at.tugraz.tc.cyfile.injection.ActivityComponent;
import at.tugraz.tc.cyfile.injection.DaggerActivityComponent;

public abstract class BaseActivity extends AppCompatActivity {

    private ActivityComponent mActivityComponent;


    public ActivityComponent getActivityComponent() {
        if (mActivityComponent == null) {
            mActivityComponent = DaggerActivityComponent.builder()
                    .applicationComponent(CyFileApplication.get(this).getComponent())
                    .build();
        }
        return mActivityComponent;
    }
}

File: CyFile-Android/app/src/main/kotlin/at/tugraz/tc/cyfile/ui/BaseActivity.kt
package at.tugraz.tc.cyfile.ui

import android.support.v7.app.AppCompatActivity

import at.tugraz.tc.cyfile.CyFileApplication
import at.tugraz.tc.cyfile.injection.ActivityComponent
import at.tugraz.tc.cyfile.injection.DaggerActivityComponent

abstract class BaseActivity : AppCompatActivity() {

    private lateinit var mActivityComponent: ActivityComponent


    val activityComponent: ActivityComponent
        get() {
            mActivityComponent = DaggerActivityComponent.builder()
                    .applicationComponent(CyFileApplication[this].component)
                    .build()
            return mActivityComponent
        }
}
